bookkeeper-server/src/test/java/org/apache/bookkeeper/client/TestRackawareEnsemblePlacementPolicy.java

```diff
@@ -525,17 +526,89 @@ public void testNewEnsembleWithMultipleRacks() throws Exception {
         addrs.add(addr4);
         repp.onClusterChanged(addrs, new HashSet<BookieSocketAddress>());
         try {
-            ArrayList<BookieSocketAddress> ensemble = repp.newEnsemble(3, 2, 2, null, new HashSet<>());
-            int numCovered = getNumCoveredWriteQuorums(ensemble, 2);
+            int ensembleSize = 3;
+            int writeQuorumSize = 2;
+            int acqQuorumSize = 2;
+            ArrayList<BookieSocketAddress> ensemble = repp.newEnsemble(ensembleSize, writeQuorumSize, acqQuorumSize,
+                    null, new HashSet<>());
+            int numCovered = getNumCoveredWriteQuorums(ensemble, 2, numOfAvailableRacks);
             assertTrue(numCovered >= 1 && numCovered < 3);
-            ArrayList<BookieSocketAddress> ensemble2 = repp.newEnsemble(4, 2, 2, null, new HashSet<>());
-            numCovered = getNumCoveredWriteQuorums(ensemble2, 2);
+            ensembleSize = 4;
+            ArrayList<BookieSocketAddress> ensemble2 = repp.newEnsemble(ensembleSize, writeQuorumSize, acqQuorumSize,
+                    null, new HashSet<>());
+            numCovered = getNumCoveredWriteQuorums(ensemble2, 2, numOfAvailableRacks);
             assertTrue(numCovered >= 1 && numCovered < 3);
         } catch (BKNotEnoughBookiesException bnebe) {
             fail("Should not get not enough bookies exception even there is only one rack.");
         }
     }
 
+    @Test
+    public void testWriteQuorumNumberOfRacks() throws Exception {
```

Comment:
can you add a unit test case to cover `write-quorum-size` is larger than `number-of-racks`?

Reply:
added testscenario